src/java/com/unifina/signalpath/AbstractSignalPathModule.java

```diff
@@ -649,7 +650,7 @@ else if (request.getType().equals("paramChange")) {
 				}
 
 				PermissionService permissionService = Holders.getApplicationContext().getBean(PermissionService.class);
-				if (!permissionService.canWrite(request.getUser(), request.getCanvas())) {
+				if (!permissionService.check(request.getUser(), request.getCanvas(), Permission.Operation.CANVAS_INTERACT)) {
```

Comment:
For `paramChange` actually, it might make sense to require `CANVAS_EDIT`. The logic is this: If you can change a parameter when the canvas is NOT running (as permitted by `CANVAS_EDIT`), you should be allowed to change it when it IS running.